src/Jackalope/ObjectManager.php

```diff
@@ -869,6 +869,7 @@ public function save()
                 if ($item->isModified() || $item->isMoved()) {
                     $item->confirmSaved();
                 }
+                $item->setClean();
```

Comment:
i think we on purpose mark the node dirty when we confirm it its saved. there are autocreated properties in the backend, e.g. the uuid if you did not force creation already.


i wrote about the solution that i expect to be correct in a comment that is now hidden:

looking at the stack trace in https://travis-ci.org/jackalope/jackalope-doctrine-dbal/jobs/119277489 i think the problem is something else. https://github.com/jackalope/jackalope/blob/check_state_delete_no_exception/src/Jackalope/ObjectManager.php#L1124 assumes that a node that is dirty is not deleted (because both are states). but then ->getIdentifier will trigger checkState which detects the node is deleted. i think before the deleted check in the object manager, we should check for dirty and if the node is dirty call refresh($keepChanges). then the state will not be dirty anymore but deleted or new or modified and things work as expected i'd hope.